src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsEndpoint.cs

```diff
@@ -166,45 +166,70 @@ public async Task<RazorDiagnosticsResponse> Handle(RazorDiagnosticsParams reques
 
             var filteredDiagnostics = unmappedDiagnostics
                 .Where(d =>
-                    // TODO: undesired HTML Diagnostic codes
-                    // (https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1257401) ||
-                    !InAttributeContainingCSharp(d, sourceText, syntaxTree, processedAttributes))
+                    !InAttributeContainingCSharp(d, sourceText, syntaxTree, processedAttributes) &&
+                    !FilterDiagnosticBasedOnErrorCode(d, sourceText, syntaxTree))
                 .ToArray();
 
             return filteredDiagnostics;
+        }
 
-            static bool InAttributeContainingCSharp(
-                Diagnostic d,
-                SourceText sourceText,
-                RazorSyntaxTree syntaxTree,
-                Dictionary<TextSpan, bool> processedAttributes)
+        private static bool FilterDiagnosticBasedOnErrorCode(Diagnostic d, SourceText sourceText, RazorSyntaxTree syntaxTree)
+        {
+            if (!d.Code.HasValue)
+            {
+                return false;
+            }
+
+            return d.Code.Value.String switch
             {
-                // Examine the _end_ of the diagnostic to see if we're at the
-                // start of an (im/ex)plicit expression. Looking at the start
-                // of the diagnostic isn't sufficient.
-                var absoluteIndex = d.Range.End.GetAbsoluteIndex(sourceText);
+                "HTML0204" => IsInvalidNestingWarningWithinComponent(d, sourceText, syntaxTree),
+                _ => false,
+            };
+
+            static bool IsInvalidNestingWarningWithinComponent(Diagnostic d, SourceText sourceText, RazorSyntaxTree syntaxTree)
+            {
+                var absoluteIndex = d.Range.Start.GetAbsoluteIndex(sourceText);
                 var change = new SourceChange(absoluteIndex, 0, string.Empty);
                 var owner = syntaxTree.Root.LocateOwner(change);
```

Comment:
Assuming most error codes will require the "owner" it may be better to just pass in the owner to this method so it's re-used

Reply:
Was considering that, however, since most error codes aren't filtered computing the owner in advance may end up being less efficient. Will abstract it out to method though to reduce duplication.